### PR TITLE
CONTRIBUTING: update page, fix Markdown syntax

### LINKED ISSUE #4986: Document how to use the tldr command against a local git repo
If this is documented anywhere, I could not find it.

Many users will install tldr using a package manager at the recommendation of a colleague. Eventually they will find a command/page that is missing and will visit the [CONTRIBUTING.md](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md). There they will find lots of information about how to create and submit a new page. They will not find any information about how to test/use their new page with the tldr command prior to submitting and getting it accepted. This is likely to reduce the number of contributors to this project and the quality of submissions. I uncovered the answer to this question by running `sudo dtruss $(which tldr) tar` but in my experience it is very rare that higher level engineers are familiar with these tools ([even though they should](https://jvns.ca/strace-zine-v3.pdf)).

Ultimately what I did was `mv ~/.tldrc/tldr-master ~/.tldrc/tldr-master.original; ln -s $PWD ~/.tldrc/tldr-master` from my the directory where I cloned this repo. This could easily be made into a script. If there is an appetite for that I can make a PR for one.

### PR DESCRIPTION
**Pull request by kbdharun**
## Changes

- Update Markdown syntax throughout the page to conform to Markdownlint.
- Minor fixes to line/paragraph spacing and grammatical errors.
- Add the requirement that an English page should be added before creating a translation.
- Suggest adding entries to translation templates when creating a new language.
- Add documentation on running `tldr-lint` locally and how to install it. (While this would seem redundant with the style guide [the reference here is a bit short], I think it is fine to refer to tldr-lint in both places (as a majority of contributors would go through this page in detail then the style guide, I will add it as a check to PR templates too in a PR today)
- Add documentation on how to batch suggestions and make a single commit.
- Add a suggestion for contributors to enable the "Allow edits by maintainers" option in PRs.
- Add more examples for commit titles and suggest <https://www.conventionalcommits.org> for other cases (i.e. non-page additions)
- Add tip about running test script via the precommit hook.

Closes #4986

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,84 +91,126 @@ refer to the [style guide](contributing-guides/style-guide.md).
...
-  - For related changes to several pages: `grep, find, locate: synchronize format of wildcards`
+Where `<command>` is the name of the command being modified, and `<type of change>` is one of the following examples:
+
+- For a new page addition: `ls: add page`, `docker-container-rm: add alias page`
+- For a page edit: `cat: fix typo`, `git-push: add --force example`
+- For a new translation of an existing page: `cp: add Tamil translation`
+- For a modification to the translation of an existing page: `cp: fix typo in Tamil translation`
+- For related changes to several pages: `grep, find, locate: synchronize format of wildcards`
+- For multiple subcommand page additions: `git-{add, push, ...}: add page`
+- For modfiying multiple pages in a language: `pages.<locale>/*: update pages`
```
Comment (kbdharun) [inline CONTRIBUTING.md:204]:
```suggestion
- For modifying multiple pages in a language: `pages.<locale>/*: update pages`
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -18,29 +18,30 @@
...
+> When submitting a new command, please base your PR against the `main` branch and check if there's already a pull request in progress for it.
 
 ## Guidelines
 
 The basic format of a `tldr` page is a set of concrete usage examples.
+
 Here are a few guidelines to get started:
 
-1. Try to keep pages at around 5 examples. Pages can be longer or shorter when appropriate, but don't exceed 8 examples.
+1. Try to keep pages at around five examples. Pages can be longer or shorter when appropriate, but don't exceed a maximum of eight examples.
```
[the remark this answers is not in the document]
Comment (kbdharun) [reply CONTRIBUTING.md:36]: Thanks, let me know once your review is complete. I will batch all your suggestions.

**Review by gutjuri — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -84,84 +91,126 @@ refer to the [style guide](contributing-guides/style-guide.md).
...
 Where possible, use inclusive language in the content of pages. For example, prefer terms like "denylist"/"allowlist" instead of "blacklist"/"whitelist", "primary"/"secondary" instead of "master"/"slave", "they" instead of "him"/"her", etc.
-Of course, this shouldn't sacrifice content clarity, such as when documenting tools where this terminology has specific technical meanings, and its usage is central to explaining the involved concepts.
+Of course, this shouldn't sacrifice content clarity, such as when documenting tools where this terminology has specific technical meanings and its usage is central to explaining the involved concepts.
 
 ## Submitting a pull request
 
-The easiest way to submit a change is to just edit the page directly on the GitHub interface.
+### Testing pages locally
+
+Once you have written a `tldr` page, you can test it's syntax locally using [`tldr-lint`](https://github.com/tldr-pages/tldr-lint).
```
[the remark this answers is not in the document]
Comment (kbdharun) [reply CONTRIBUTING.md:151]: Sure.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -18,29 +18,30 @@
 [license-image]: https://img.shields.io/badge/license-CC_BY_4.0-blue.svg?label=License
 
 Contributions to the tldr-pages project are [most welcome](GOVERNANCE.md)!
-All `tldr` pages are stored in Markdown right here on GitHub.
-Just open an issue or send a pull request and we'll incorporate it as soon as possible.
+
+All `tldr` pages are stored in Markdown right here on GitHub. Just open an issue or send a pull request, we'll incorporate it as soon as possible.
```
Comment (sbrl) [inline CONTRIBUTING.md:22]: Perhaps this will read better? 
```suggestion
All `tldr` pages are stored in Markdown right here on GitHub. Just open an issue or send a pull request, and we'll incorporate it as soon as possible.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -18,29 +18,30 @@
...
+> When submitting a new command, please base your PR against the `main` branch and check if there's already a pull request in progress for it.
 
 ## Guidelines
 
 The basic format of a `tldr` page is a set of concrete usage examples.
+
 Here are a few guidelines to get started:
 
-1. Try to keep pages at around 5 examples. Pages can be longer or shorter when appropriate, but don't exceed 8 examples.
+1. Try to keep pages at around five examples. Pages can be longer or shorter when appropriate, but don't exceed the maximum of eight examples.
```
Comment (sbrl) [inline CONTRIBUTING.md:36]: This isn't necessary. We're not writing a formal formal formal document here.
```suggestion
1. Try to keep pages at around 5 examples. Pages can be longer or shorter when appropriate, but don't exceed the maximum of eight examples.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -18,29 +18,30 @@
...
    Remember, it's OK if the page doesn't cover everything; that's what `man` is for.
 2. When in doubt, keep new command-line users in mind. Err on the side of clarity rather than terseness.
    For example, commands that require `sudo` should include it directly in the examples.
-3. Try to incorporate the spelled-out version of single-letter options in the example's description.
+3. Try incorporating the spelled-out version of single-letter options in the example's description.
```
Comment (sbrl) [inline CONTRIBUTING.md:40]: This read better in the original form.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -67,12 +68,18 @@ As a quick reference, the format of each page should match the following templat
 `command --option1 --option2 {{arg_value}}`
 ```
 
+For page descriptions, you can additionally use ``See also: `command`.`` and [subcommand mention](#subcommands).
```
Comment (sbrl) [inline CONTRIBUTING.md:71]: Not sure what `subcommand mention` makes grammatical sense here.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -84,84 +91,129 @@ refer to the [style guide](contributing-guides/style-guide.md).
...
+The latest version of [NodeJS](https://nodejs.org) is required to install `tldr-lint` with the following command:
+
+```sh
+npm install --global tldr-lint
+```
+
+Once its installed, you can test your page by running the following command:
+
+```sh
+tldr-lint <path/to/page.md>
```
Comment (sbrl) [inline CONTRIBUTING.md:162]: Shouldn't we use the `{{token_syntax}}` here? :P
```suggestion
tldr-lint {{path/to/page.md}}
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -84,84 +91,129 @@ refer to the [style guide](contributing-guides/style-guide.md).
...
 
-The easiest way to apply suggested changes is to accept the suggestion made on your pull request. Refer to the [GitHub docs](https://docs.github.com/en/enterprise-server@3.2/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) for more details.
+The easiest way to apply suggested changes is to accept the suggestion made on your pull request. Refer to the [GitHub docs](https://docs.github.com/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) for more details.
 
-To commit a suggestion to your pull request, simply click on `Commit suggestion`:
+To commit a suggestion to your pull request, click on `Commit suggestion`:
 
 ![Commit suggestion button in Github](./images/commit-suggestion-button.png)
 
+When you want to commit multiple suggestions, go to the "Files changed" tab and batch all suggestions. Now, click `Commit suggestions` button and enter a commit message to create a single commit.
```
Comment (sbrl) [inline CONTRIBUTING.md:191]: Perhaps this will read better?
```suggestion
If you want to commit multiple suggestions, go to the "Files changed" tab and batch all suggestions. Now, click `Commit suggestions` button and enter a commit message to create a single commit.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -84,84 +91,129 @@ refer to the [style guide](contributing-guides/style-guide.md).
...
 
-    <command>: type of change
+`<command>: type of change`
 
-Examples:
-  - For a new page addition: `ls: add page`
-  - For a page edit: `cat: fix typo`, `git-push: add --force example`
-  - For a new translation of an existing page: `cp: add Tamil translation`
-  - For related changes to several pages: `grep, find, locate: synchronize format of wildcards`
+Where `<command>` is the name of the command being modified, and `<type of change>` is one of the following examples:
```
Comment (sbrl) [inline CONTRIBUTING.md:199]: Extra rogue angle brackets here.
```suggestion
Where `<command>` is the name of the command being modified, and `type of change` is one of the following examples:
```

...but this seems quite prescriptive and restrictive in the kinds of commit messages people can use here. I'd suggest more broadly rewording this bit.

**Review by sbrl — CHANGES_REQUESTED** [review]
Thanks for edit, @kbdharun! I've left some comments below for you to review.

**Review by sbrl — APPROVED** [review]
Looks good to me! Thanks for the hard work, @kbdharun <3
